src/graph/graph.c

```diff
@@ -512,39 +538,33 @@ void Graph_CreateNode(Graph *g, int label, Node *n) {
 	en->prop_count = 0;
 	en->properties = NULL;
 	n->entity = en;
+	_Graph_AddNodeToLabelMatrix(g, id, label);
+}
 
+inline void Graph_TryAddLabelMatrix(Graph *g, int label) {
 	if(label != GRAPH_NO_LABEL) {
```

Comment:
I suggest inverting this into a guard clause:
`if(label == GRAPH_NO_LABEL) return;`